src/components/App.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from "react";
+import LocationDetails from "./LocationDetails";
+import ForecastSummaries from "./ForecastSummaries";
+import ForecastDetails from "./ForecastDetails";
+import getForecast from "../requests/getForecast";
+import SearchForm from "./SearchForm";
+import "../styles/App.css";
+
+const App = () => {
+  const [forecasts, setForecasts] = useState([]);
+  const [location, setLocation] = useState({ city: "", country: "" });
+  const [selectedDate, setSelectedDate] = useState(0);
+  const [searchText, setSearchText] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const selectedForecast = forecasts.find(
+    (forecast) => forecast.date === selectedDate
+  );
+
+  useEffect(() => {
+    getForecast(
+      setSelectedDate,
+      setForecasts,
+      setLocation,
+      searchText,
+      setErrorMessage
+    );
+  }, []);
+
+  const handleCitySearch = () => {
+    getForecast(
+      setSelectedDate,
+      setForecasts,
+      setLocation,
+      searchText,
+      setErrorMessage
+    );
+  };
+
+  const handleForecastSelect = (date) => {
+    setSelectedDate(date);
+  };
+
+  const { city, country } = location;
+
+  return (
+    <div className="weather-app">
+      <LocationDetails
+        city={city}
+        country={country}
+        errorMessage={errorMessage}
+      />
+      <SearchForm
+        onSubmit={handleCitySearch}
+        searchText={searchText}
+        setSearchText={setSearchText}
+      />
+      {!errorMessage && (
```

Comment:
Have you tried to deliberately make the app fail by passing it a nonsense string, then tried to enter a valid city?
If not, try it. You'll see that the app will never leave the error screen. Why is this?

The components are rendered on the condition that the `errorMessage` is empty, but once one has been created, there is no logic in place to remove it even if a successful request to get more forecasts is made! Thankfully we can fix that by calling `setErrorMessage` whenever a new request is made to the API